compmbox/compress.c

```diff
@@ -929,15 +929,15 @@ static int comp_path_pretty(struct Buffer *buf, const char *folder)
 /**
  * comp_path_parent - Find the parent of a Mailbox path - Implements MxOps::path_parent() - @ingroup mx_path_parent
  */
-static int comp_path_parent(char *buf, size_t buflen)
+static int comp_path_parent(struct Buffer *path)
 {
-  if (mutt_path_parent(buf))
+  if (mutt_path_parent(path->data))
```

Comment:
Is this correct? Do we need to call buf_fix_dptr() here?

Reply:
Yes and yes.
As we're likely to be shortening the string, we need to ensure `dptr` isn't left pointing beyond the `NUL`.

Reply:
ok, that's what I thought